src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -547,7 +547,11 @@ private static PropertyBag<ObjectTypeProperty> PopulatePropertyBag(SerializableO
             ObjectTypeProperty? additionalPropertiesProperty = null;
             ObjectTypeProperty? rawDataField = null;
 
-            var inheritedDictionarySchema = inputModel.GetSelfAndBaseModels().OfType<InputDictionaryType>().FirstOrDefault();
+            // TODO - The following warning is being reported on L554:
```

Comment:
I'm not sure how to fix this one!

Reply:
Hi since we are more familiar with the code in this repo, how about we take over the task of updating this to dotnet 8?

Reply:
Passing a null might not be our initial intention here, but this is truly how this code would produce.
I am trying to fix this to be correct in my PR.